lib/datasets.go

```diff
@@ -935,20 +935,24 @@ func (m *DatasetMethods) Pull(p *PullParams, res *dataset.Dataset) error {
 
 	ctx := context.TODO()
 
-	ref, source, err := m.inst.ParseAndResolveRef(ctx, p.Ref, "network")
+	if p.Remote == "" {
+		p.Remote = "network"
```

Comment:
Is it a problem to modify the input parameters? We don't usually do that. Might be worth creating a local var instead.

Reply:
shouldn't be, but you're right it's a bad practice. Will change